### PR TITLE
test: Generalize tmpdir and file/dir backup/restoration

### PR DESCRIPTION
**Pull request by martinpitt**

### CODE REVIEW

**Comment by martinpitt** [comment]
[This virsh backend failure](https://logs.cockpit-project.org/logs/pull-13751-20200318-183128-c7e8b44f-ubuntu-1804/log.html#77) could be relevant: [screenshot](https://logs.cockpit-project.org/logs/pull-13751-20200318-183128-c7e8b44f-ubuntu-1804/TestMachinesVirsh-testAddDisk-ubuntu-1804-127.0.0.2-2201-FAIL.png) shows a permission error.

File: test/common/testlib.py [hunk 1]
```diff
@@ -1219,6 +1218,43 @@ def sed_file(self, expr, path, apply_change_action=None):
                 self.addCleanup(m.execute, apply_change_action)
             self.addCleanup(m.execute, "mv {0}.cockpittest {0}".format(path))
 
+    def restore_dir(self, path, post_restore_action=None):
```
Comment (martinpitt) [inline test/common/testlib.py]: Let's add a `reboot=` argument; if true, it uses classical `cp -a` instead of bind mounts.

Comment (marusak) [reply test/common/testlib.py]: Implemented in https://github.com/cockpit-project/cockpit/pull/13758/commits/f71ef5bd8a4bf5ecc389df3a838defe4f638518b

**Comment by martinpitt** [comment]
Several failures of [testDetachDisk](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13751-20200319-151710-94cad161-debian-testing/log.html), but that has been brittle for a while outside of that PR. It works fine locally, so for now I think that's an unrelated flake.

**Comment by marusak** [comment]
[testDetachDisk](https://logs.cockpit-project.org/logs/pull-13751-20200320-072721-94cad161-debian-testing/log.html) succeeded. I think we can call this flake. (although rather intrusive together with testNICAdd)

**Pull request merged**